gulp-tasks/lint.js

```diff
@@ -1,35 +1,67 @@
-'use strict';
-
-// Include gulp
-const { src } = require('gulp');
-
-// Include Our Plugins
+const { src, lastRun } = require('gulp');
+const { formatters } = require("stylelint");
 const gulpStylelint = require('gulp-stylelint');
 const eslint = require('gulp-eslint');
+const dutil = require('./utils/doc-util');
+const cFlags = require("./utils/cflags");
 
-// Export our tasks.
-module.exports = {
+const IGNORE_STRING = "This file is ignored";
 
-  // Lint Sass based on .stylelintrc.yml config.
-  lintSass: function () {
-    return src(['./src/patterns/stylesheets/**/*.scss'])
-      .pipe(
-        gulpStylelint({
-          reporters: [
-            {
-              formatter: 'string',
-              console: true
-            }
-          ]
-        })
+function ignoreStylelintIgnoreWarnings(lintResults) {
+  return formatters.string(
+    lintResults.reduce((memo, result) => {
+      const { warnings } = result;
+      const fileIsIgnored = warnings.some((warning) =>
+        RegExp(IGNORE_STRING, "i").test(warning.text)
       );
-  },
-
-  // Lint JavaScript based on .eslintrc config.
-  lintJS: function() {
-    return src(['src/patterns/**/**/*.js'])
-      .pipe(eslint({ fix: true }))
-      .pipe(eslint.format())
-      .pipe(eslint.failAfterError());
+
+      if (!fileIsIgnored) {
+        memo.push(result);
+      }
+
+      return memo;
+    }, [])
+  );
+}
+
+// Lint Sass based on .stylelintrc.json config.
+function lintSass() {
+  const stylelintOptions = {
+    failAfterError: true,
+    reporters: [
+      {
+        formatter: ignoreStylelintIgnoreWarnings,
+        console: true,
+      },
+    ],
+    syntax: "scss",
+  };
+
+  return src(
+    "./src/patterns/stylesheets/**/*.scss",
+    { since: lastRun(lintSass) }
```

Comment:
Had to move `lintSass()` to it's own function to be able to use [lastRun](https://gulpjs.com/docs/en/api/lastrun/). Reduces wait times for linting from an average of `11s → 3s`